services/microsoft.storage/resource.ftl

```diff
@@ -5,15 +5,18 @@
         AZURE_STORAGE_SERVICE : {
             AZURE_STORAGEACCOUNT_RESOURCE_TYPE : {
                 "apiVersion" : "2019-04-01",
-                "type" : "Microsoft.Storage/storageAccounts"
+                "type" : "Microsoft.Storage/storageAccounts",
+                "conditions" : [ "name_to_lower" ] [#-- TODO: Impliment conditions, and add resourceId() functionality --]
```

Comment:
To define and document the available conditions it might be good to look at the compositeObject set. This would be used when you are adding sevices to the resource profile to set default values and show available ones 

stackOutput has a good example of this 
https://github.com/codeontap/gen3/blob/8580ab5da9e7313d0e78762cad9ca2b7414eb356/engine/inputdata/stackOutput.ftl 

Instead of the += or each service you would instead call a macro - for stack outputs this is 
```
addStackOutputs
``` 

It then looks after adding to the global variable while also performing some validation based on the configuration 
```
stackOutputConfiguration
``` 

https://github.com/codeontap/gen3/blob/8580ab5da9e7313d0e78762cad9ca2b7414eb356/engine/inputdata/stackOutput.ftl#L152 

This is handy since you then don't need to add the default [] conditions to every resource and it will yell if you haven't set an apiVersion or type which would both be mandatory in this case

Reply:
@roleyfoley  I have addressed this now in the [latest commit](https://github.com/codeontap/gen3-azure/pull/40/commits/eef9877dfc6e0bb688057bb076d44dde1b41dd7e).